machine/mtrap.c

```diff
@@ -18,7 +18,7 @@
 
 void __attribute__((noreturn)) bad_trap(uintptr_t* regs, uintptr_t dummy, uintptr_t mepc)
 {
-  die("machine mode: unhandlable trap %d @ %p", read_csr(mcause), mepc);
+  die("machine mode: unhandlable trap %ld @ %p", read_csr(mcause), (void*)mepc);
```

Comment:
Is this cast needed?

Reply:
Clang warns otherwise.

Reply:
Ugh, you're right, uintptr_t isn't a %p (presumably it's allowed to have a different representation to void*?), you'd have to use PRIxPTR etc instead. So yeah, this is fine...